### ISSUE TITLE
Quest journal don't refresh if login on different map [Fixed] + New quest "talk to" no completely implemented.

### ISSUE THREAD

**Issue by Xentar** [suriyun-production/mmorpg-kit-docs#674]
Hey Suri!

Noticed this issue now, but tested an old version i had installed and was there too, so is no exclusive to last version.
If you take a quest from Map001, and go to another map, logout and close client, when you start client again and log, the quest don't appear.  (if you relog without close client, the quest appear... my guess, same session when you took it)
If you return to the map where the quest was given, it appear back on the quest log. 

Usually i search the cause and post it too, but this time... no sure the reason.
--- 
[1 day later]
Ok. Found the reason and the way to fix it.
The problem is that there is no a "quests database". When the npcDialog of an NPC is on the map, he populate the scriptable into the gameinstance... so, if there is no npc, no population.
A easy fix is create an quest database scriptable object and make the gameinstance load them on the start... Of course, this should be on GameDatabase like the npc's, classes, etc. But as a fix, work for now.

Cheers!
P.D.: wish you can left the quest system more open, i want to expand it but is the less partial class out there, jejeje. There is no even a talk to npc option as quest... something more scriptableobject for example =_= Well, just a suggestion taking adventage of that i find a bug on the quest system hehehe

**Comment by insthync** [comment]
I've brought `Quests` config back to `GameInstance` in 1.60c update, you can set quest data to that list to make it loaded and stored to the game instance when start.

I also add `Talk To Npc` quest task type, you can try it.

**Comment by Xentar** [comment]
Great! Updated last night and fixed some prefabs that disconected.
Now got to test the new quest additions.
First i must thanks you a lot for this changes.

Now: The questDatabase works (deleted my own fix)
But the talk to Npc its giving me trouble. Probably i'm doing it wrong. Created 1 quest and two dialogs. 
-The quest have the Npc Entity who need to be talked and the quest dialog that going to appear once i talked with him.
-The first dialog is the quest giver while the second dialog, the dialog of the quest
The first thing i notice, is that on quest journal the quest task is 0/0. When i talk with the npc, it run the quest dialog fine, but the quest is not completed.

Did all this on the Map001 of a clean project.
I did something wrong or something is missing?

Noticed a "custom" quest type too. Its for future adition or can be "linked" to a custom event? if is the case, how it works?

Thanks in advance, @insthync
